marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -281,8 +261,33 @@ private void executeWalk(
         }
     }
 
-    private DocumentMetadataHandle buildMetadata(String collection, String permissions) {
+    /**
+     * As of 5.2.0, artifact permissions are separate from module permissions. If artifact permissions
+     * are not defined, then it falls back to using default permissions.
+     *
+     * @param collection
+     * @param config
+     * @return
+     */
+    protected DocumentMetadataHandle buildMetadata(HubConfig config, String collection) {
         DocumentMetadataHandle meta = new DocumentMetadataHandle();
+        String permissions = null;
+        switch (collection) {
+            case "http://marklogic.com/entity-services/models":
```

Comment:
Same thing - avoid a switch statement by having the caller pass in the permissions string instead of a HubConfig.